notch8/atla_digital_library
# CSV imports- format_digital field parsing

co-author: @alishaevn 

# Story
After bulkrax upgrade, on CSV imports, the format (digital) field was showing the original imported MIME types instead of the terms that were set up to be used by ATLA in their [formats.yml](https://github.com/scientist-softserv/atla_digital_library/blob/main/config/authorities/formats.yml) authority. 

Refs #231 

# Expected Behavior Before Changes
Format (digital) field was not being parsed, and was being saved with just the MIME type data from src.
For example, if the data in the csv format digital column said `audio/mpeg`, the work's format field was not being parsed, and would be saved as `audio/mpeg`

# Expected Behavior After Changes
After importing CSVs with bulkrax, the format field should render the correctly parsed format term that corresponds with its mime type. in the file [formats.yml](https://github.com/scientist-softserv/atla_digital_library/blob/main/config/authorities/formats.yml), the "alts" are the MIME types that would be entered from src, and the "term" is the format field that should be saved/rendered.

# Screenshots / Video
<img width="773" alt="Screen Shot 2023-04-27 at 9 29 22" src="https://user-images.githubusercontent.com/73361970/234894671-7e2ecbda-1cbe-47f1-ad68-d613b9dce79b.png">

# Testing Instructions
1. Go to dashboard on staging : [dl-staging](https://dl-staging.atla.com/)
2. Update/re-run the following importer: https://dl-staging.atla.com/importers/64?locale=en
3. make sure you reharvest all items, i.e. remove them all and re-create them. this is the bottom button option when updating the importer
4. After the works are done importing, check on the work show page that the format (digital) of the items in the importer say "MPEG" and not "audio/mpeg"

# Notes
- its also possible that the date field was not being parsed correctly for csvs since it was not included in the parsed_fields array of the #process_parse method. we added it there to ensure it is being parsed correctly as well. there are no other custom methods that did not already have fields included in this array.
